File: app/src/main/java/id/ac/uasbungaa/data/local/DatabaseHelper.java
package id.ac.uasbungaa.data.local;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import kotlin.jvm.internal.DefaultConstructorMarker;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class DatabaseHelper extends SQLiteOpenHelper {
    private static final String DATABASE_NAME = "UAS200020016";
    private static final int DATABASE_VERSION = 2;
    private static final String CREATE_TABLE_USER = "CREATE TABLE USER_200020016 (_id INTEGER PRIMARY KEY AUTOINCREMENT, username TEXT NOT NULL, password TEXT NOT NULL, thumbnail TEXT NOT NULL)";
    private static final String CREATE_TABLE_BARANG = "CREATE TABLE BARANG_200020016 (_id INTEGER PRIMARY KEY AUTOINCREMENT, nama TEXT NOT NULL, kode TEXT NOT NULL, stok INTEGER NOT NULL, harga INTEGER NOT NULL, gambar TEXT NOT NULL)";
    @NotNull
    public static final DatabaseHelper.Companion Companion = new DatabaseHelper.Companion((DefaultConstructorMarker) null);

    public void onCreate(@Nullable SQLiteDatabase db) {
        if (db != null) {
            db.execSQL(CREATE_TABLE_USER);
        }

        if (db != null) {
            db.execSQL(CREATE_TABLE_BARANG);
        }

    }

    public void onUpgrade(@Nullable SQLiteDatabase db, int oldVersion, int newVersion) {
        if (db != null) {
            db.execSQL("DROP TABLE IF EXISTS USER_200020016");
        }

        if (db != null) {
            db.execSQL("DROP TABLE IF EXISTS BARANG_200020016");
        }

        this.onCreate(db);
    }

    public DatabaseHelper(@NotNull Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    public static final class Companion {
        private Companion() {
        }

        public Companion(DefaultConstructorMarker $constructor_marker) {
            this();
        }
    }
}

File: app/src/main/java/id/ac/uasbunga/data/local/DatabaseHelper.kt
package id.ac.uasbunga.data.local

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.PASSWORD
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.TABLE_NAME
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.THUMBNAIL
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.USERNAME
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion._ID

internal class DatabaseHelper(context: Context) :
    SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    companion object {
        private const val DATABASE_NAME = "UAS200020016"
        private const val DATABASE_VERSION = 2
        private const val CREATE_TABLE_USER = "CREATE TABLE $TABLE_NAME" +
                " (${_ID} INTEGER PRIMARY KEY AUTOINCREMENT," +
                " ${USERNAME} TEXT NOT NULL," +
                " ${PASSWORD} TEXT NOT NULL," +
                " ${THUMBNAIL} TEXT NOT NULL)"
        private const val CREATE_TABLE_BARANG =
            "CREATE TABLE ${DatabaseContract.BARANG_200020016.TABLE_NAME}" +
                    " (${DatabaseContract.BARANG_200020016._ID} INTEGER PRIMARY KEY AUTOINCREMENT," +
                    " ${DatabaseContract.BARANG_200020016.NAMA} TEXT NOT NULL," +
                    " ${DatabaseContract.BARANG_200020016.KODE} TEXT NOT NULL," +
                    " ${DatabaseContract.BARANG_200020016.STOK} INTEGER NOT NULL," +
                    " ${DatabaseContract.BARANG_200020016.HARGA} INTEGER NOT NULL," +
                    " ${DatabaseContract.BARANG_200020016.GAMBAR} TEXT NOT NULL)"
    }

    override fun onCreate(db: SQLiteDatabase?) {
        db?.execSQL(CREATE_TABLE_USER)
        db?.execSQL(CREATE_TABLE_BARANG)
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        db?.execSQL("DROP TABLE IF EXISTS $TABLE_NAME")
        db?.execSQL("DROP TABLE IF EXISTS ${DatabaseContract.BARANG_200020016.TABLE_NAME}")
        onCreate(db)
    }
}